plugins/modules/proxmox_backup.py

```diff
@@ -0,0 +1,758 @@
+  pool:
+    description:
+      - Backup all known guest systems included in the specified pool.
+      - Can not be use with O(vmid) and O(pool) in same job.
+    type: str
+  protected:
+    description:
+      - If V(true), mark backup(s) as protected.
+    type: bool
+  prune_backups:
+    description:
+      - Use these retention options instead of those from the storage configuration.
+    type: str
+  quiet:
+    description:
+      - Be quiet.
+    type: bool
+  remove:
+    description:
+      - Prune older backups according to O(prune_backups).
+    type: bool
+  repeat_missed:
+    description:
+      - If V(true), the job will be run as soon as possible if it was missed while the scheduler was not running.
+    type: bool
+  schedule:
+    description:
+      - Backup schedule. The format is a subset of `systemd` calendar events.
+    type: str
+  script:
+    description:
+      - Use specified hook script.
+    type: str
+  starttime:
+    description:
+      - Job Start time.
+    type: str
+  stdexcludes:
+    description:
+      - Exclude temporary files and logs.
+    type: bool
+  stop:
+    description:
+      - Stop running backup jobs on this host.
+    type: bool
+  stopwait:
+    description:
+      - Maximal time to wait until a guest system is stopped (minutes).
+    type: int
+  storage:
+    description:
+      - Store resulting file to this storage.
+    type: str
+  tmpdir:
+    description:
+      - Store temporary files to specified directory.
+    type: str
+  vmid:
+    description:
+      - The ID of the guest system you want to backup.
+      - Can not be use with vmid and pool in same job
+    type: str
+  zstd:
+    description:
+      - Zstd threads. V(N=0) uses half of the available cores,
+        if V(N) is set to a value bigger than V(0), V(N) is used as thread count.
+    type: int
+
+extends_documentation_fragment:
+  - community.general.proxmox.actiongroup_proxmox
+  - community.general.proxmox.documentation
+  - community.general.attributes
+'''
+
+EXAMPLES = '''
+- name: List all backup jobs
+  community.general.proxmox_backup:
+    api_host: "node1"
+    api_user: user@realm
+    api_password: password
+    validate_certs: false
+    state: list
+  register: backup_result
+
+- name: Show current backup job
+  ansible.builtin.debug:
+    var: backup_result
+
+- name: Create backup with id backup-20bad73a-d245
+  community.general.proxmox_backup:
+    api_host: "node1"
+    api_token_id: "token_id"
+    api_token_secret: "token_secret"
+    validate_certs: false
+    id: "backup-20bad73a-d245"
+    vmid: "103"
+    mode: "snapshot"
+    mailnotification: "always"
+    mailto: "my mail address"
+    repeat_missed: 0
+    enabled: 1
+    prune_backups:
+        keep_yearly: "6"
+        keep_weekly: "5"
+        keep_hourly: "4"
+        keep_daily: "2"
+        keep_last: "1"
+        keep_monthly: "3"
+    storage: "backup-idcheck-preprod-0"
+    schedule: "*-*-* 22:00:00"
+    state: present
```

Comment:
I would also add an example of how to use `notes_template`, since this is far from trivial:
```suggestion
    state: present
    # To be able to use {{ }} in a notes_template without Ansible interpreting them, mark the string as unsafe:
    notes_template: !unsafe >-
      Backup of {{cluster}}'s {{node}} VM {{vmid}}.
```